cnf-tests/testsuites/e2esuite/dpdk/dpdk.go

```diff
@@ -274,6 +276,187 @@ sleep INF
+			netAnnotationPodRX = fmt.Sprintf(
+				`[{"name": "%s", "mac": "%s", "namespace": "%s"}, {"name": "%s", "interface": "%s", "namespace": "%s"},
+						{"name": "%s", "interface": "%s", "namespace": "%s"}, {"name": "%s", "interface": "%s", "namespace": "%s"}]`,
+				dpdkNetwork1, DPDK_SERVER_WORKLOAD_MAC, namespaces.DpdkTest, tapNetwork, tapIface, namespaces.DpdkTest,
+				tapVlan100Network, tapVlanIface100, namespaces.DpdkTest, tapVlan200Network, tapVlanIface200, namespaces.DpdkTest)
+
+			// Set container_use_devices SELinux Boolean if disable.
+			isCUDDisabled = isContainerUseDevicesSEBooleanDisabled(node)
+			if isCUDDisabled {
+				setContainerUseDevicesSEBoolean(node)
+			}
+
+			// Create sriov resources.
+			networks.CreateDpdkPolicy(sriovDevice, node, dpdkResource1, "#0-1", 5, true)
+			networks.CreateDpdkPolicy(sriovDevice, node, dpdkResource2, "#2-4", 5, false)
+			networks.WaitStable(sriovclient)
+			Eventually(func() int64 {
+				testedNode, err := sriovclient.Nodes().Get(context.Background(), node, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				resNum, _ := testedNode.Status.Allocatable[corev1.ResourceName("openshift.io/"+dpdkResource1)]
+				capacity, _ := resNum.AsInt64()
+				return capacity
+			}, 10*time.Minute, time.Second).Should(Equal(int64(2)))
+			Eventually(func() int64 {
+				testedNode, err := sriovclient.Nodes().Get(context.Background(), node, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				resNum, _ := testedNode.Status.Allocatable[corev1.ResourceName("openshift.io/"+dpdkResource2)]
+				capacity, _ := resNum.AsInt64()
+				return capacity
+			}, 10*time.Minute, time.Second).Should(Equal(int64(3)))
+
+			networks.CreateSriovNetwork(sriovclient, sriovDevice, dpdkNetwork1, namespaces.DpdkTest, namespaces.SRIOVOperator, dpdkResource1, "")
+			networks.CreateSriovNetworkWithVlan(sriovclient, sriovDevice, dpdkNetwork2, namespaces.DpdkTest, namespaces.SRIOVOperator, dpdkResource2, "", vlan100)
+
+			// Create tap and vlan NADs.
+			nadTap, err = networks.NewNetworkAttachmentDefinitionBuilder(namespaces.DpdkTest, tapNetwork).WithTap().Build()
+			Expect(err).ToNot(HaveOccurred())
+			err = client.Client.Create(context.Background(), nadTap)
+			Expect(err).ToNot(HaveOccurred())
+
+			nadVlan100, err = networks.NewNetworkAttachmentDefinitionBuilder(namespaces.DpdkTest, tapVlan100Network).WithVlan(tapIface, vlan100, true).WithHostLocalIpam("1.1.1.0").Build()
+			Expect(err).ToNot(HaveOccurred())
+			err = client.Client.Create(context.Background(), nadVlan100)
+			Expect(err).ToNot(HaveOccurred())
+
+			nadVlan200, err = networks.NewNetworkAttachmentDefinitionBuilder(namespaces.DpdkTest, tapVlan200Network).WithVlan(tapIface, vlan200, true).WithHostLocalIpam("1.1.1.0").Build()
+			Expect(err).ToNot(HaveOccurred())
+			err = client.Client.Create(context.Background(), nadVlan200)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		AfterEach(func() {
```

Comment:
this part is problematic has today it will run before we run the reporter this means we are not going to see anything if this test fails because the afterEach will remove everything before the reporter have the opportunity to collect the info.

Until we start using the new method to collect the info I think is better just to have another `it` to clean everything

Reply:
ok, will change it

Reply:
but Ginkgo expects the specs to be independent to be randomized, right?

Reply:
Yes, I think you have to add the `Ordered` label to the Context
https://onsi.github.io/ginkgo/#ordered-containers

Reply:
sure, but it's better to change all tests in this file once the new method is adopted. So, I will leave it like this.